chaos_tests/chaos_test.go

```diff
@@ -0,0 +1,449 @@
+		}
+		return value, nil
+	}
+
+	// Register the workflow
+	dbos.RegisterWorkflow(dbosCtx, recvWorkflow)
+
+	err := dbosCtx.Launch()
+	require.NoError(t, err)
+
+	// Run multiple workflows with send/recv
+	numWorkflows := 10000
+	for i := range numWorkflows {
+		if i%100 == 0 {
+			t.Logf("Starting workflow %d/%d", i+1, numWorkflows)
+		}
+		handle, err := dbos.RunWorkflow(dbosCtx, recvWorkflow, "")
+		require.NoError(t, err, "failed to start workflow %d", i)
+
+		// Generate a random value
+		value := uuid.NewString()
+
+		// Give some time to the workflow to enter the sleep state
+		time.Sleep((5 * time.Millisecond))
+
+		// Send the value to the workflow
+		err = dbos.Send(dbosCtx, handle.GetWorkflowID(), value, topic)
+		require.NoError(t, err, "failed to send value for workflow %d", i)
+
+		// Get the result and verify it matches
+		result, err := handle.GetResult()
+		require.NoError(t, err, "failed to get result for workflow %d", i)
+		assert.Equal(t, value, result, "unexpected result for workflow %d", i)
+	}
+}
+
+// Test event functionality
+func TestChaosEvents(t *testing.T) {
+	dbosCtx := setupDBOS(t)
+
+	// Start chaos monkey
+	var wg sync.WaitGroup
+	defer wg.Wait()
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	PostgresChaosMonkey(t, ctx, &wg)
+
+	key := "test_key"
+
+	// Define event workflow
+	eventWorkflow := func(ctx dbos.DBOSContext, _ string) (string, error) {
+		value := uuid.NewString()
+		err := dbos.SetEvent(ctx, key, value)
+		if err != nil {
+			return "", fmt.Errorf("failed to set event: %w", err)
+		}
+		return value, nil
+	}
+
+	// Register the workflow
+	dbos.RegisterWorkflow(dbosCtx, eventWorkflow)
+
+	err := dbosCtx.Launch()
+	require.NoError(t, err)
+
+	// Run multiple workflows with events
+	numWorkflows := 5000
+	for i := range numWorkflows {
+		if i%100 == 0 {
+			t.Logf("Starting workflow %d/%d", i+1, numWorkflows)
+		}
+		wfID := uuid.NewString()
+
+		// Start workflow with specific ID
+		handle, err := dbos.RunWorkflow(dbosCtx, eventWorkflow, "", dbos.WithWorkflowID(wfID))
+		require.NoError(t, err, "failed to start workflow %d", i)
+
+		// Get the workflow result
+		value, err := handle.GetResult()
+		require.NoError(t, err, "failed to get result for workflow %d", i)
+
+		// Retrieve the event and verify it matches
+		retrievedValue, err := dbos.GetEvent[string](dbosCtx, wfID, key, 0)
+		require.NoError(t, err, "failed to get event for workflow %d", i)
+		assert.Equal(t, value, retrievedValue, "unexpected event value for workflow %d", i)
+	}
+}
+
+// Test queue functionality
+func TestChaosQueues(t *testing.T) {
+	dbosCtx := setupDBOS(t)
+
+	// Start chaos monkey
+	var wg sync.WaitGroup
+	defer wg.Wait()
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	PostgresChaosMonkey(t, ctx, &wg)
+
+	queue := dbos.NewWorkflowQueue(dbosCtx, "test_queue")
+
+	// Define step functions
+	stepOne := func(ctx dbos.DBOSContext, x int) (int, error) {
```

Comment:
The other thing we could check is if a scheduled workflow keeps ticking after reconnection